tests/src/lib.rs

```diff
@@ -11,197 +11,9 @@ extern crate pwasm_test;
-	assert!(!endpoint.inner.called_wrong, "wrong method was invoked");
-}
-
-#[test]
-fn sam_dispatch() {
-	#[derive(Default)]
-	struct TestContractInstance {
-		called: bool,
-		called_wrong: bool,
-	}
-
-	impl TestContract for TestContractInstance {
-		fn constructor(&mut self, _p1: bool) {
-		}
-		fn sam(&mut self, p1: Vec<u8>, p2: bool, p3: Vec<U256>) {
-			assert_eq!(p1, vec![100, 97, 118, 101]);
-			assert_eq!(p2, true);
-			assert_eq!(p3, vec![
-				U256::from("1"),
-				U256::from("2"),
-				U256::from("3"),
-			]);
-			self.called = true;
-		}
-		fn baz(&mut self, _p1: u32, _p2: bool) {
-			self.called_wrong = true;
-		}
-		fn boo(&mut self, _arg: u32) -> u32 {
-			self.called_wrong = true;
-			0
-		}
-	}
-
-	let mut endpoint = TestEndpoint::new(TestContractInstance::default());
-	let result = endpoint.dispatch(PAYLOAD_SAMPLE_2);
-
-	assert_eq!(result, Vec::new());
-
-	assert!(endpoint.inner.called, "`sam` method was not invoked");
-	assert!(!endpoint.inner.called_wrong, "wrong method was invoked");
-}
-
-#[test]
-fn boo_dispatch() {
-	#[derive(Default)]
-	struct TestContractInstance {
-		called: bool,
-		called_wrong: bool,
-	}
-
-	impl TestContract for TestContractInstance {
-		fn constructor(&mut self, _p1: bool) {
-		}
-		fn sam(&mut self, _p1: Vec<u8>, _p2: bool, _p3: Vec<U256>) {
-			self.called_wrong = true;
-		}
-		fn baz(&mut self, _p1: u32, _p2: bool) {
-			self.called_wrong = true;
-		}
-		fn boo(&mut self, arg: u32) -> u32 {
-			self.called = true;
-			assert_eq!(arg, 69);
-			255
-		}
-	}
-
-	let mut endpoint = TestEndpoint::new(TestContractInstance::default());
-	let result = endpoint.dispatch(PAYLOAD_SAMPLE_3);
-
-	assert_eq!(&result[28..32], &[0x00, 0x00, 0x00, 0xff]);
-
-	assert!(endpoint.inner.called, "`boo` method was not invoked");
-	assert!(!endpoint.inner.called_wrong, "wrong method was invoked");
-}
-
-#[allow(dead_code)]
-mod contract {
-	use pwasm_abi_derive::eth_abi;
-
-	#[eth_abi(EmptyEndpoint, _EmptyClient)]
-	trait EmptyContract {
-		fn constructor(&mut self, _p: bool);
-	}
-
-	pub struct EmptyContractInstance {
-		pub called: bool,
-		pub p: bool
-	}
-
-	impl EmptyContract for EmptyContractInstance {
-		fn constructor(&mut self, p1: bool) {
-			self.called = true;
-			self.p = p1;
-		}
-	}
-}
-
-#[test]
-fn dispatch_empty_abi() {
-
-	let mut endpoint = contract::EmptyEndpoint::new(contract::EmptyContractInstance{called: false, p: false});
-	endpoint.dispatch_ctor(&[0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01]);
-	assert!(endpoint.inner.called);
-	assert!(endpoint.inner.p);
-}
-
-#[test]
-fn baz_call() {
-	ext_reset(|e| e.endpoint(Address::zero(), Endpoint::ok()));
-	let mut client = Client::new(Address::zero());
-	client.baz(69, true);
-	assert_eq!(ext_get().calls()[0].input.as_ref(), PAYLOAD_SAMPLE_1);
-}
+mod detail;
```

Comment:
why is it called `detail` ? :)

Reply:
Good question. To me it looked like a very detailed test case. :D 
I am very open to better name suggestions. I just wanted to separate library definition from an actual test.

Reply:
could be `general` or `simple`

Reply:
okay I will rename it to `general`

Reply:
done